### PR TITLE
[RFC] Inline function declaration with single arg as object

### LINKED ISSUE #1164: [Flow] Don't indent single function param with multiline type annotation
Using [0.22.0](https://prettier.github.io/prettier/#%7B%22content%22%3A%22const%20X%20%3D%20(props%3A%20%7B%5Cn%20%20a%3A%20boolean%2C%5Cn%7D)%20%3D%3E%5Cn%20%20%3CA%20%2F%3E%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22all%22%2C%22bracketSpacing%22%3Afalse%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22flow%22%2C%22doc%22%3Afalse%7D%7D)
```es6
const X = (props: {
  a: boolean,
}) =>
  <A />;

// is formatted to

const X = (
  props: {
    a: boolean,
  },
) => <A />;
```
It would be great if we kept the param inline, the same way we do for single param destructing.

### PR DESCRIPTION
**Pull request by vjeux**
Fixes #1164

### CODE REVIEW

**Comment by vjeux** [comment]
Some of the examples look weird. I'm not 100% sure that it's an improvement.

**Comment by xixixao** [comment]
@vjeux This looks great to me. Can we ship it? It's the only current big blocker I know for our codebase.

**Comment by vjeux** [comment]
Okay, I finally pinpointed what I was uneasy about. https://github.com/prettier/prettier/issues/1409 Has the same issue where the return type breaks instead of the arguments (which are inlined via the same heuristic).

Something changes in the way things are grouped and the order of breaking is different.

We probably need to add or remove a group() such that the arguments break before the return type. Once it's fixed, we can merge this one :)

File: tests/flow/dictionary/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -43,9 +43,10 @@ function foo0(
   return x;
 }
 
-function foo2(
-  x: { [key: string]: number }
-): { [key: string]: number, +toString: () => string } {
+function foo2(x: { [key: string]: number }): {
+  [key: string]: number,
+  +toString: () => string
+} {
```
Comment (xixixao) [inline tests/flow/dictionary/__snapshots__/jsfmt.spec.js.snap]: Can you add
```es6
function foo2(x: {
  [key: string]: number,
  toString: () => string,
}): { [key: string]: number } {}
```
and
```es6
function foo2(x: {
  [key: string]: number,
}): { [key: string]: number } {}
```
to the tests? The return type should not be split over multiple lines if it has only on key in the type.

**Comment by vjeux** [comment]
Okay, I fixed the root cause in https://github.com/prettier/prettier/pull/1453. I'm going to be able to push this forward :)

**Pull request merged**